src/avlizer_confluent.erl

```diff
@@ -303,10 +303,18 @@ terminate(_Reason, _State) ->
 
 %%%_* Internals ================================================================
 
-get_registry_url() ->
-  {ok, #{ schema_registry_url := URL
-        }} = application:get_env(?APPLICATION, ?MODULE),
-  URL.
+get_registry_base_request() ->
+  case application:get_env(?APPLICATION, ?MODULE) of
+    {ok, #{schema_registry_url := URL, schema_registry_auth := Auth}} -> 
+      {URL, get_registry_headers_closure(Auth)};
+    {ok, #{schema_registry_url := URL}} -> 
+      {URL, get_registry_headers_closure()}
+  end.
+
+get_registry_headers_closure({basic, Username, PasswordClosure}) -> 
```

Comment:
I don't see the point of having a closure to protect function call args passing around internally.
was @k32 your idea to have a closure in app-env? I wonder in what cases does Erlang (crash) log app-env.

Reply:
@k32 what do you think? Maybe we get rid of headers' closure and keep the password env closure, to at least don't expose the password without `base64`? Or else can we get rid of both closures? I'm either fine with both decisions...

Reply:
When it comes to avlizer, just follow what @zmstone says :)

Reply:
Perfect, done! :v: